proposals/texture-component-swizzle.md

```diff
@@ -0,0 +1,93 @@
+# Texture component swizzle 🥤
+
+**Roadmap:** This proposal is **under active development, but has not been standardized for inclusion in the WebGPU specification. The proposal is likely to change before it is standardized.** WebGPU implementations **must not** expose this functionality; doing so is a spec violation. Note however, an implementation might provide an option (e.g. command line flag) to enable a draft implementation, for developers who want to test this proposal.
+
+Issue:
+
+- https://github.com/gpuweb/gpuweb/issues/5179
+
+
+## Motivation
+
+Texture component swizzle lets you to specify how the channels of a texture (red, green, blue, and alpha) are mapped to the color components when accessed by a shader. It is supported everywhere WebGPU runs, except for Mac family 1 GPU devices. Therefore it would be nice to expose it as an optional feature.
+
+## API changes
+
+The GPUFeatureName `"texture-component-swizzle"` controls access to texture component mapping, which is anticipated to be available in Metal, Vulkan, D3D12, and OpenGL ES 3.
+
+```webidl
+partial enum GPUFeatureName {
+    "texture-component-swizzle",
+};
+```
+
+The GPUTextureViewDescriptor `swizzle` option lets you rearrange or even replace the data from a texture's channels when creating a GPUTextureView. This is done by assigning a value from the GPUComponentSwizzle enum to each of the `r`, `g`, `b`, and `a` components in your texture view.
+
+```webidl
+partial dictionary GPUTextureViewDescriptor {
+    GPUTextureComponentSwizzle swizzle;
+};
+
+// Structure specifying a custom color component mapping for a texture view.
+dictionary GPUTextureComponentSwizzle {
+    GPUComponentSwizzle r = "identity";
```

Comment:
Is `identity` necessary and/or beneficial? One could write this as:

```
    GPUComponentSwizzle r = "r";
    GPUComponentSwizzle g = "g";
    GPUComponentSwizzle b = "b";
    GPUComponentSwizzle a = "a";
```

Reply:
FYI I've just sent https://github.com/gpuweb/gpuweb/pull/5227